### ISSUE TITLE
It's near impossible to bundle edge adapters

### ISSUE THREAD

**Issue by steveklabnik** [jruby/activerecord-jdbc-adapter#68]
Hey Nick-

With the way the code is currently laid out, you can't really bundle any of the adapters. There's a few things that contribute to this, and I'd like some feedback before I work up some sort of pull request:
### The issues
1. The adapters themselves require the exact version of the main library. This means that I can't just bundle edge of `activerecord-jdbc-adapter`, I also _have_ to bundle edge `activerecord-jdbcpostgresql-adapter` too. This could be fixed by relaxing [this](https://github.com/nicksieger/activerecord-jdbc-adapter/blob/master/adapters/postgresql/activerecord-jdbcpostgresql-adapter.gemspec#L25) to `~>1.1.2`. 
2. It's impossible to tell bundler to use the edge adapters from a git repository. This is because bundler expects sub-gems to be in a directory with their name, right under top level, like Rails does. I [made a tiny commit on my fork](https://github.com/steveklabnik/activerecord-jdbc-adapter/commit/05cb099b1be92fc8fb43f78e193eb8f8e5e30a49), but this doesn't take into account any of the Rakefiles, etc. But it does let you bundle properly, with 
   
    git "https://github.com/steveklabnik/activerecord-jdbc-adapter.git" do
      gem "activerecord-jdbc-adapter", :ref => "whatever"
      gem "activerecord-jdbcpostgresql-adapter"
    end

Which is cool! But this leads to problem #3:
1. Because you're using `hoe`, I think, the built jar and the Mainfest.txt are in .gitignore. This leads to messages like
   
   Using activerecord-jdbc-adapter (1.1.1) from git://github.com/steveklabnik/activerecord-jdbc-adapter.git (at master)
   activerecord-jdbc-adapter at blahblahblah did not have a valid gemspec.
   This prevents bundler from installing bins or native extensions, but that may not affect its functionality.
   The validation message from Rubygems was:
   ["Manifest.txt", "lib/arjdbc/jdbc/adapter_java.jar"] are not files

Then, it doesn't actually work, and gives you 

```
no such file to load -- arjdbc/jdbc/adapter_java
```

which is expected.

Now, I'm not sure if this is fix-able. It makes sense that binaries aren't kept in git, but is there a way we can get Bundler to generate them?
### What I had to do

To get it to work, I ended up pulling down my fork, shoving it in vendor/gems, and removing the .git folder from the directory, and then committing it all in my app. I know that some people 'Vendor Everything,' but that really bothers me... Point is, this isn't so much an urgent thing as a "I thought I'd let you know about it" thing.
### In conclusion

I'd love to hear some feedback about this problem, and if there's anything that can be done to fix it. I do think that moving the adapters out to the top level would be nice, but it's your project, you can organize it as you wish. If that interests you, I'd be willing to work up a patch to change all the paths in all the support files.
